lib/lte_link_control/lte_lc_helpers.c

```diff
@@ -22,6 +22,112 @@
 
 LOG_MODULE_REGISTER(lte_lc_helpers, CONFIG_LTE_LINK_CONTROL_LOG_LEVEL);
 
+static K_MUTEX_DEFINE(list_mtx);
+
+/**@brief List element for event handler list. */
+struct event_handler {
+	sys_snode_t        node;
+	lte_lc_evt_handler_t handler;
+};
+
+static sys_slist_t handler_list;
+
+/**
+ * @brief Find the handler from the event handler list.
+ *
+ * @return The node or NULL if not found and its previous node in @p prev_out.
+ */
+static struct event_handler *event_handler_list_find_node(struct event_handler **prev_out,
```

Comment:
This mechanism seems very useful.  You could make it a separate library for others to use for other purposes, not just the LTE LC event handler.

Reply:
That's true. I don't think we have the time to do it at the moment and these LTE LC changes were already put into a PR to get them to master. The PR is now reviewed and ready and just about to get merged.